libs/shared/src/entities/notification-template/notification-template.interface.ts

```diff
@@ -36,6 +36,17 @@ export interface INotificationTrigger {
   identifier: string;
   variables: INotificationTriggerVariable[];
   subscriberVariables?: INotificationTriggerVariable[];
+  snippetVariables?: ITriggerSnippetVariable[];
+}
+
+export enum TriggerContextTypeEnum {
+  TENANT = 'tenant',
+  ACTOR = 'actor',
+}
```

Comment:
I will remove the actor behviour from this PR, but just wanted to show the direction I intend to take here. It will make adding new sections of trigger objects easier and more immediate